src/neptune/versionHelper.d

```diff
@@ -0,0 +1,334 @@
+/*******************************************************************************
+
+    Helper to deal with version strings
+
+    Copyright:
+        Copyright (c) 2017 Sociomantic Labs GmbH. All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE.txt for details.
+
+*******************************************************************************/
+
+module neptune.versionHelper;
+
+import std.algorithm.iteration : splitter;
+import octod.api.repos : Tag;
+
+/// Structure to easily deal with version strings
+struct Version
+{
+    import std.typecons : Nullable;
+
+    /// Type of the version
+    enum Type { Major, Minor, Patch, Invalid };
```

Comment:
This seems redundant information to me. Major version is always the one with minor and patch numbers set to 0. Minor one is always with just patch set to 0. If all numbers are present, it is a patch version,